api/src/pcapi/core/users/factories.py

```diff
@@ -65,6 +65,8 @@ def _create(
         kwargs["password"] = crypto.hash_password(password)
         instance = super()._create(model_class, *args, **kwargs)
         instance.clearTextPassword = password
+        cls.on_init(instance)
+        repository.save(instance)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Pourquoi on a soudain besoin de ça ?

Reply:
Si tu parles spécifiquement de `repository.save`, très bonne question. Je ne me l'explique pas. Mais s'il n'y est pas ça ne marche pas.
À l'occasion je serai curieux de creuser le sujet.

Reply:
Yes c'est bien ça ma question :D

Reply:
En fait rien de bien étonnant : 

Dans la classe `SQLAlchemyModelFactory` qui est la classe parente de `BaseFactory`, la méthode `_create`appelle la méthode classe `_save`qui fait bien un `session.add(obj)`et `session.commit()`.

Ici comme on fait les modifications après le `_create`, eh bien, on doit refaire le `add`et `commit`.